### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by olga-bylnova**

### CODE REVIEW

File: core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java [hunk 1]
```diff
@@ -9,9 +10,15 @@
...
-public class JdbcPhoneDao implements PhoneDao{
+public class JdbcPhoneDao implements PhoneDao {
     @Resource
     private JdbcTemplate jdbcTemplate;
+    private static final String FIND_ALL_SQL =
+            "SELECT * FROM (SELECT * FROM phones OFFSET %d LIMIT %d) as phones_partial " +
+                    "LEFT JOIN phone2color on phones_partial.id = phone2color.phoneId " +
+                    "LEFT JOIN colors on colors.id = phone2color.colorId";
+
+    private final ProductRowCallbackHandler<Phone> handler = new PhoneRowCallbackHandler();
```
Comment (edragalev) [inline core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java]: 1. Does that thread-save?. That object has a state, which will be accessed by all threads which use that DAO
2. Check what is happening on the website when you reload productListPage

File: core/pom.xml [hunk 2]
```diff
@@ -12,6 +12,11 @@
   <packaging>jar</packaging>
   <version>1.0-SNAPSHOT</version>
   <dependencies>
+    <dependency>
+      <groupId>com.es.phoneshop</groupId>
+      <artifactId>phoneshop-demodata</artifactId>
```
Comment (edragalev) [inline core/pom.xml]: Does it running fine on your local? For me app is not running with that circular dependency here

Comment (olga-bylnova) [reply core/pom.xml]: It actually didn't, but I added this dependency for tests and tests were running fine and I didn't check the overall application. I'll fix that, thanks for your remark.

**Comment by edragalev** [comment]
Check where is your test folder for core module located
https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html

**Review by edragalev — APPROVED** [review]

**Pull request merged**